packages/cli/src/utils/changelog.js

```diff
@@ -34,16 +80,37 @@ const getChangelogFromMarkdown = (version, markdown) => {
   endingLine -= 1;
 
   const changelogLines = lines.slice(startingLine, endingLine);
+  const changelog = [];
+  const appMetadata = [];
+  const issueMetadata = [];
+
+  for (const line of changelogLines) {
+    let changeType = null;
+    if (startsWithIgnoringSymbols(line, 'update')) {
+      changeType = 'FEATURE_UPDATE';
+    } else if (startsWithIgnoringSymbols(line, 'fix')) {
+      changeType = 'BUGFIX';
+    }
```

Comment:
I thought about this example:

```
## 1.0.0
1. Fix trigger/new_card #123 
2. Update and fix search/find_contact #567 
3. You're gonna like this - improve create/add_task #89
4. Add trigger/new_contact #10
5. New action! create/add_contact
```

Can we adjust the logic so the above example outputs:

```javascript
// app metadata
[
  {"change_type": "BUGFIX", "action_type": "read", "action_key": "new_card"},

  // BUGFIX because "fix" is right next to search/find_contact, closer than "update"
  {"change_type": "BUGFIX", "action_type": "search", "action_key": "find_contact"},

  // The word "improve" doesn't have to be in the beginning of the sentence
  {"change_type": "FEATURE_UPDATE", "action_type": "write", "action_key": "add_task"},

  {"change_type": "FEATURE_UPDATE", "action_type": "read", "action_key": "new_contact"},

  {"change_type": "FEATURE_UPDATE", "action_type": "write", "action_key": "add_contact"},
];

// issue metadata
[
  {"change_type": "BUGFIX", "issue_id": 123},

  // BUGFIX because "fix" is closer to the issue ID 
  {"change_type": "BUGFIX", "issue_id": 567},

  // The word "improve" doesn't have to be in the beginning of the sentence
  {"change_type": "FEATURE_UPDATE", "issue_id": 89},

  {"change_type": "FEATURE_UPDATE", "issue_id": 10},
];
```

These are all the verbs/words I think we can map:

- BUGFIX
  - fix/fixes/fixed
- FEATURE_UPDATE
  - add/adds/added
  - improve/improves/improved
  - update/updates/updated
  - new

If you can think of anything else, feel free to add them to the list!

Reply:
So I partially implemented this without the lookback.

It could handle lookback by collecting "orphaned" metadata then yielding that once it can determine the context, but that feels more likely to catch false positives and increase complexity.